### PR TITLE
Mack Update Project Application Layer and Endpoint

### PR DESCRIPTION
**Pull request by MackMcCall**
- Created UpdateProjectByGuid Request and Handler
- Added UpdateProjectByGuid Endpoint
- Created tests for the request and handler

### CODE REVIEW

**Review by JalenMAshanti — APPROVED** [review]
Nice

File: TicketMate.Application/Requests/ProjectRequests/UpdateByGuid/UpdateProjectByGuidRequest.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        public bool? IsActive { get; set; }
+
+
+        public bool IsValid(out Validator validator)
+        {
+            validator = new();
+
+            validator.ApplyRule(new GuidRequiredRule(Guid, nameof(Guid)));
+            if (Name != null)
+                validator.ApplyRule(new StringLengthLimitRule(Name, nameof(Name), MaxLength.ProjectName));
```
Comment (uhyeay2) [inline TicketMate.Application/Requests/ProjectRequests/UpdateByGuid/UpdateProjectByGuidRequest.cs]: Can we achieve the same result by passing isRequiredString = false into the constructor like below?
```
new StringLengthLimitRule(Name, nameof(Name), MaxLength.ProjectName, isRequiredString = false);
```

Your tests for the request object looked pretty good to me, so I think if you implement this change you can just re-run your tests to make sure it's applying the rule as expected.

**Review by uhyeay2 — APPROVED** [review]

**Review by BraydonTGS — APPROVED** [review]
I LOVE YOU

File: TicketMate.Application/Requests/ProjectRequests/UpdateByGuid/UpdateProjectByGuidRequest.cs (hunk 1 above)
Comment (uhyeay2) [reply TicketMate.Application/Requests/ProjectRequests/UpdateByGuid/UpdateProjectByGuidRequest.cs]: Resolving Comment as Mack pushed up this change - glad it worked out good @MackMcCall !

**Pull request merged**